pkg/async/notifications/implementations/gcp_processor.go

```diff
@@ -0,0 +1,91 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	"github.com/NYTimes/gizmo/pubsub"
+	"github.com/flyteorg/flyteadmin/pkg/async"
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/golang/protobuf/proto"
+)
+
+// TODO: Add a counter that encompasses the publisher stats grouped by project and domain.
+type GcpProcessor struct {
+	sub           pubsub.Subscriber
+	email         interfaces.Emailer
+	systemMetrics processorSystemMetrics
+}
+
+func NewGcpProcessor(sub pubsub.Subscriber, emailer interfaces.Emailer, scope promutils.Scope) interfaces.Processor {
+	return &GcpProcessor{
+		sub:           sub,
+		email:         emailer,
+		systemMetrics: newProcessorSystemMetrics(scope.NewSubScope("gcp_processor")),
+	}
+}
+
+func (p *GcpProcessor) StartProcessing() {
+	for {
+		logger.Warningf(context.Background(), "Starting GCP notifications processor")
+		err := p.run()
+		logger.Errorf(context.Background(), "error with running GCP processor err: [%v] ", err)
+		time.Sleep(async.RetryDelay)
+	}
+}
+
+func (p *GcpProcessor) run() error {
```

Comment:
nit : can we pass in the context and use it in places where we are using context.Background